kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsTlsIT.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+
+import javax.net.ssl.SSLHandshakeException;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.testcontainers.DockerClientFactory;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import io.kroxylicious.kms.provider.hashicorp.vault.VaultResponse.ReadKeyData;
+import io.kroxylicious.kms.provider.hashicorp.vault.config.Config;
+import io.kroxylicious.proxy.config.tls.FilePassword;
+import io.kroxylicious.proxy.config.tls.InlinePassword;
+import io.kroxylicious.proxy.config.tls.InsecureTls;
+import io.kroxylicious.proxy.config.tls.Tls;
+import io.kroxylicious.proxy.config.tls.TrustStore;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
+
+/**
+ * Integration tests for HashiCorp Vault with TLS, checking that we can make
+ * API calls with TLS enabled on the server.
+ */
+class VaultKmsTlsIT {
+
+    private static final String VAULT_TOKEN = "token";
+    private TestVault vaultContainer;
+
+    private static final CertificateGenerator.Keys keys = CertificateGenerator.generate();
+
+    @BeforeEach
+    void beforeEach() {
+        assumeThat(DockerClientFactory.instance().isDockerAvailable()).withFailMessage("docker unavailable").isTrue();
```

Comment:
I wonder if there is a better way to do this. I have a feeling there is some sort of `DisabledIf` type annotation. Anyway not for this PR.